File: Recursion and Backtracking/Fibonacci_Numbers.cpp
#include <iostream>

using namespace std;

int fibonacci(int n) {
    if(n==0) return 0;
    if(n==1) return 1;
    return fibonacci(n-1)+fibonacci(n-2);

}

int main() {
    int n;
    cin >> n;
    cout << fibonacci(n);
    return 0;
}

File: Recursion and Backtracking/Fibonacci_numbers.py
cache = {0:0, 1:1}

def fibonacci(n):
    try:
        return cache[n]
    except KeyError:
        cache[n] = fibonacci(n - 1) + fibonacci(n - 2)
        return cache[n]

n = int(input())
print(fibonacci(n))
